spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -244,14 +244,8 @@ public DistributionMode positionDeleteDistributionMode() {
         return deleteMode;
       }
     } else {
-      // use hash distribution if write distribution is range or hash and table is partitioned
-      // avoid range-based shuffles unless the user asks explicitly
-      DistributionMode writeMode = distributionMode();
-      if (writeMode != NONE && table.spec().isPartitioned()) {
-        return HASH;
-      } else {
-        return writeMode;
-      }
+      // avoid range-based shuffles for partitioned tables
+      return table.spec().isPartitioned() ? HASH : RANGE;
```

Comment:
My current thinking is that we should probably request some distribution by default. If the table is partitioned, we should cluster by spec ID and partition to avoid the range-based shuffle. In all other cases, it is probably better to ask for an ordered distribution by spec ID, partition, file and position to reduce the number of produced delete files. Having a lot of delete files will probably cost us more than doing some work during the DELETE operation.